services/event-application-service/src/pubsub_controller/connection_factory.h

```diff
@@ -0,0 +1,37 @@
+#ifndef PUBSUB_CONTROLLER_CONNECTION_FACTORY_H
+#define PUBSUB_CONTROLLER_CONNECTION_FACTORY_H
+
+#include <memory>
+
+#include "google/cloud/pubsub/message.h"
+#include "google/cloud/pubsub/subscriber_connection.h"
+#include "google/cloud/pubsub/subscriber_options.h"
+#include "google/cloud/pubsub/subscription.h"
+#include "google/cloud/pubsub/publisher_connection.h"
+#include "google/cloud/pubsub/publisher_options.h"
+
+namespace eas::pubsub_controller {
+class IConnFactory {
+ public:
+  virtual std::shared_ptr<google::cloud::pubsub::SubscriberConnection>
+  MakeSubscriberConnection(google::cloud::pubsub::Subscription subscription,
+                           google::cloud::Options opts) = 0;
+  virtual std::shared_ptr<google::cloud::pubsub::PublisherConnection>
+  MakePublisherConnection(google::cloud::pubsub::Topic topic,
+                          google::cloud::Options opts) = 0;
+
+  virtual ~IConnFactory(){};
+};
+
+class PubSubConnFactory final : public IConnFactory {
+ public:
+  std::shared_ptr<google::cloud::pubsub::SubscriberConnection>
+  MakeSubscriberConnection(google::cloud::pubsub::Subscription subscription,
+                           google::cloud::Options opts = {}) override;
+  std::shared_ptr<google::cloud::pubsub::PublisherConnection>
+  MakePublisherConnection(google::cloud::pubsub::Topic topic,
+                          google::cloud::Options opts = {}) override;
+};
```

Comment:
Can you also add this here?
```
namespace pubsub = ::google::cloud::pubsub;
namespace cloud = ::google::cloud;
```
These nested namespaces barely readable
Or since you need only options, subscription, connection and topic, you can add some typedef for them

Reply:
Unfortunately no. I can't use using in class definitions.

But typedef can be a good idea, I will consider this. Thanks